tests/components/husqvarna_automower/test_lawn_mower.py

```diff
@@ -0,0 +1,133 @@
+"""Tests for lawn_mower module."""
+import logging
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from aioautomower.model import MowerAttributes, MowerList
+from aioautomower.session import AutomowerSession
+import pytest
+
+from homeassistant.components.husqvarna_automower import DOMAIN
+from homeassistant.components.lawn_mower import LawnMowerActivity
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from .common import setup_integration
+from .utils import make_mower_list
+
+from tests.common import MockConfigEntry, load_fixture, load_json_value_fixture
+
+_LOGGER = logging.getLogger(__name__)
+
+TEST_MOWER_ID = "c7233734-b219-4287-a173-08e3643f89f0"
+
+
+@pytest.mark.enable_socket
+@pytest.mark.asyncio
+async def setup_entity(hass: HomeAssistant, mowers):
+    """Set up entity and config entry."""
+
+    config_entry: MockConfigEntry = await setup_integration(hass)
+
+    config_entry.add_to_hass(hass)
+    token_decoded = load_fixture("token_decoded.json", DOMAIN)
+    with patch(
+        "aioautomower.session.AutomowerSession",
+        return_value=AsyncMock(
+            name="AutomowerMockSession",
+            model=AutomowerSession,
+            data=mowers,
+            register_data_callback=MagicMock(),
+            unregister_data_callback=MagicMock(),
+            connect=AsyncMock(),
+            resume_schedule=AsyncMock(),
+        ),
+    ), patch(
+        "homeassistant.helpers.config_entry_oauth2_flow.async_get_config_entry_implementation",
```

Comment:
We shouldn't need to patch all these details.

Reply:
If it isn't patched, there are errors like that: `FAILED tests/components/husqvarna_automower/test_lawn_mower.py::test_lawn_mower_commands - TypeError: <AsyncMock name='async_get_config_entry_implementation().async_refresh_token()' id='139843651950352'>`
I don't know another way to avoid them.